rectools/dataset/interactions.py

```diff
@@ -189,5 +200,7 @@ def to_external(
             res[Columns.Weight] = self.df[Columns.Weight]
         if include_datetime:
             res[Columns.Datetime] = self.df[Columns.Datetime]
+        if include_extra_cols:
+            self._add_extra_cols(res, self.df.drop(columns=[Columns.Weight, Columns.Datetime]))
```

Comment:
The `.drop` operation is extremely heavy since it creates a copy of the dataframe.  
It worths to write some additional code to avoid using it

Reply:
fixed

Reply:
fixed again